src/circuits/merkle_sum_tree.rs

```diff
@@ -78,243 +77,10 @@ impl <F:Field> Circuit<F> for MerkleSumTreeCircuit<F> {
             )?;
         }
 
-        // compute the sum of the merkle sum tree as sum of the leaf balance and the sum of the path elements balances
-        let computed_sum = self.leaf_balance + self.path_element_balances.iter().fold(F::zero(), |acc, x| acc + x);
-
-        // enforce computed sum to be less than the assets sum 
-        chip.enforce_less_than(layouter.namespace(|| "enforce less than"), &next_sum, computed_sum, self.assets_sum)?;
+        // enforce computed sum to be less than the assets sum
+        chip.enforce_less_than(layouter.namespace(|| "enforce less than"), &next_sum)?;
```

Comment:
Yeah, it is a clever approach indeed! I created a [script](https://github.com/enricobottazzi/tomaquet-ecc/blob/main/test_ecc.py#L8) on python and it seems to work as expected. A few points:

- in order to work the `LtChip` should be modified as `LeChip` meaning less or equal, otherwise a case in which `left=0` `right = 1` `sum=1` would fail the check `right < sum` 
- the performance compared to the `overflow_chip` should be benchmarked in order to decide which to use
- It should be tested carefully inside halo2, as a similar "bug" was spotted in circom => https://github.com/iden3/circomlib/pull/86